provider/pkg/provider/provider.go

```diff
@@ -2426,6 +2426,14 @@ func (k *kubeProvider) Delete(ctx context.Context, req *pulumirpc.DeleteRequest)
 			// to consider the CR to be deleted as well in this case.
 			return &pbempty.Empty{}, nil
 		}
+		if strings.HasSuffix(urn.Type().String(), "Patch") && await.IsDeleteRequiredFieldErr(awaitErr) {
+			if cause, ok := errors.StatusCause(awaitErr, metav1.CauseTypeFieldValueRequired); ok {
+				awaitErr = fmt.Errorf(
+					"this Patch resource is currently managing a required field, so it can't be deleted "+
+						"directly. Either set the `retainOnDelete` resource option, or transfer ownership of the "+
+						"field before deleting: %s", cause.Field)
```

Comment:
Did we want to add a link to some tutorial for this once we have one?

Reply:
Yeah, I'm planning to update the message once we have some better docs online